### PR TITLE
Add trafficReplayerMaxUptime parameter and functionality

### PR DESCRIPTION
**Pull request by AndreKurait**
### Description
This change adds a new deployment parameter for the trafficReplayer `trafficReplayerMaxUptime` along with functionality and documentation. This allows a user to set a max uptime after which AWS will startup a new container and kill the old one.

This achieves the functionality by adding a sidecar "MaxUptimeContainer" which succeeds a healthcheck after a given startup time and fails after a given maxUptime. This allows ECS to provision a new task and wait for the startup time before sending SIGTERM to the old task.

Shutdown Behavior: Note, shutdown logs for the existing replayer were not observed and replayer. Created task to follow up on SIGTERM behavior in [MIGRATIONS-1698](https://opensearch.atlassian.net/browse/MIGRATIONS-1698)

* Category: New Feature
* Why these changes are required? This allows customers to have more controlled behavior of the traffic replayer by limiting edge case scenarios that can arise from a long running task like out of memory errors due to leaks and the time between startup and teardown increasing
* What is the old behavior before changes and new behavior after changes? New parameter `trafficReplayerMaxUptime` added, no change to default behavior when parameter is not specified

### Issues Resolved
[MIGRATIONS-1670](https://opensearch.atlassian.net/browse/MIGRATIONS-1670)

Is this a backport? If so, please add backport PR # and/or commits #

### Testing
Deployed to AWS and verified restarting behavior. Note, an observation on shutdown behavior was made resulting in followup [MIGRATIONS-1698](https://opensearch.atlassian.net/browse/MIGRATIONS-1698)

### Check List
- [x] New functionality includes testing
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: deployment/cdk/opensearch-service-migration/lib/service-stacks/migration-service-core.ts [hunk 1]
```diff
@@ -146,6 +148,38 @@ export class MigrationServiceCore extends Stack {
...
+                        "UPTIME=$(awk '{print int($1)}' /proc/uptime); " +
+                        `test $UPTIME -gt ${startupPeriodSeconds} && ` +
+                        `test $UPTIME -lt ${maxUptimeSeconds}`
+                    ],
+                    timeout: Duration.seconds(2),
+                    retries: 1,
+                    startPeriod: Duration.seconds(startupPeriodSeconds * 2)
+                }
+            });
+            maxUptimeContainer.addContainerDependencies({
```
Comment (gregschohn) [inline deployment/cdk/opensearch-service-migration/lib/service-stacks/migration-service-core.ts:176]: can you add a comment to say which way the dependency is running?  Who depends upon whom.
What does "START" mean two lines below?

File: deployment/cdk/opensearch-service-migration/lib/service-stacks/migration-service-core.ts [hunk 2]
```diff
@@ -146,6 +148,38 @@ export class MigrationServiceCore extends Stack {
...
+                    "/bin/sh",
+                    "-c",
+                    "sleep infinity"
+                ],
+                essential: true,
+                healthCheck: {
+                    command: [
+                        "CMD-SHELL",
+                        "UPTIME=$(awk '{print int($1)}' /proc/uptime); " +
+                        `test $UPTIME -gt ${startupPeriodSeconds} && ` +
```
Comment (gregschohn) [inline deployment/cdk/opensearch-service-migration/lib/service-stacks/migration-service-core.ts:168]: why is this line needed?

File: deployment/cdk/opensearch-service-migration/lib/service-stacks/migration-service-core.ts [hunk 3]
```diff
@@ -146,6 +148,38 @@ export class MigrationServiceCore extends Stack {
...
+                image: ContainerImage.fromRegistry("public.ecr.aws/amazonlinux/amazonlinux:2023-minimal"), 
+                memoryLimitMiB: 64, 
+                entryPoint: [
+                    "/bin/sh",
+                    "-c",
+                    "sleep infinity"
+                ],
+                essential: true,
+                healthCheck: {
+                    command: [
```
Comment (gregschohn) [inline deployment/cdk/opensearch-service-migration/lib/service-stacks/migration-service-core.ts:165]: A one-line, high-level comment about using uptime of the host every time the healthcheck is called would be helpful for others that aren't familiar with ECS/

File: deployment/cdk/opensearch-service-migration/lib/service-stacks/migration-service-core.ts [hunk 4]
```diff
@@ -146,6 +148,38 @@ export class MigrationServiceCore extends Stack {
...
+                        "CMD-SHELL",
+                        "UPTIME=$(awk '{print int($1)}' /proc/uptime); " +
+                        `test $UPTIME -gt ${startupPeriodSeconds} && ` +
+                        `test $UPTIME -lt ${maxUptimeSeconds}`
+                    ],
+                    timeout: Duration.seconds(2),
+                    retries: 1,
```
Comment (gregschohn) [inline deployment/cdk/opensearch-service-migration/lib/service-stacks/migration-service-core.ts:172]: what will these mean if there's a failure?  Just bail and consider it dead?

File: deployment/cdk/opensearch-service-migration/options.md [hunk 5]
```diff
@@ -24,6 +24,7 @@ These tables list all CDK context configuration values a user can specify for th
...
 | trafficReplayerExtraArgs             | string  | "--sigv4-auth-header-service-region es,us-east-1 --speedup-factor 5"   | Extra arguments to provide to the Replayer command. This includes auth header options and other parameters supported by the [Traffic Replayer](../../../TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/TrafficReplayer.java).                                     |
+| trafficReplayerMaxUptime          | string  | "PT2H"                                                                 | The maximum uptime for the Traffic Replayer service, specified in ISO 8601 duration format. This controls how long the Traffic Replayer will run before automatically shutting down. Example values: "PT30M" (30 minutes), "PT2H" (2 hours). When this duration is reached, ECS will initiate the startup of a new Traffic Replayer task to ensure continuous operation. This mechanism ensures that the Traffic Replayer service can manage its resources effectively and prevent issues associated with long running processes. Set to the greater of the given value 5 minutes. |
```
Comment (gregschohn) [inline deployment/cdk/opensearch-service-migration/options.md:27]: 2H seems a little arbitrary.  I would say in terms of most easy to understand, hourly, daily (if it's a big problem to recycle too often) or even every 6 hours.  Doing this often enough that an operator WILL see it happen could be a plus.

Where is the default actually defined?

File: deployment/cdk/opensearch-service-migration/lib/service-stacks/migration-service-core.ts (hunk 2 above)
Comment (AndreKurait) [reply deployment/cdk/opensearch-service-migration/lib/service-stacks/migration-service-core.ts:168]: This sets the minimum time that a task should be up before considering it heatlhy. This is needed because we don't have healthchecks configured for our replayer. 

We can reduce it lower than 30 seconds, and possibly remove it, but then we increase the risk of having a few second delay between the partitions being revoked on the existing task vs assigned to the new one

File: deployment/cdk/opensearch-service-migration/lib/service-stacks/migration-service-core.ts (hunk 4 above)
Comment (AndreKurait) [reply deployment/cdk/opensearch-service-migration/lib/service-stacks/migration-service-core.ts:172]: Say /proc/uptime is unavailable or something like that, ecs, will retry the call once more, but then consider it dead. Given this container is essential, the task will be killed and restarted

File: deployment/cdk/opensearch-service-migration/options.md (hunk 5 above)
Comment (AndreKurait) [reply deployment/cdk/opensearch-service-migration/options.md:27]: There is no default currently so the deployment will not deploy the artifacts (maxUptimeContainer) needed to monitor this. 

I'll change this to discuss hourly and daily as examples

File: deployment/cdk/opensearch-service-migration/lib/service-stacks/migration-service-core.ts (hunk 1 above)
Comment (AndreKurait) [reply deployment/cdk/opensearch-service-migration/lib/service-stacks/migration-service-core.ts:176]: Added

**Review by gregschohn — APPROVED** [review]
Approved with a rec to add one more comment.  Thanks!

File: deployment/cdk/opensearch-service-migration/lib/service-stacks/migration-service-core.ts (hunk 2 above)
Comment (AndreKurait) [reply deployment/cdk/opensearch-service-migration/lib/service-stacks/migration-service-core.ts:178]: added
